### PR TITLE
Automated qa tests for install from list feature

### PR DESCRIPTION
**Pull request by amstone326**

### CODE REVIEW

File: features/install_from_list.feature [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	Then I do not see view with id "edit_email"
+	
+	Then I press view with class "android.widget.Switch" 
+	
+	# Verify that we're now in web user auth mode
+	Then I do not see view with id "edit_domain"
+	Then I see view with id "edit_email"
+
+	# Test getting app list for a web user
+	Then I auth for install from list with email "commcare.mobile.testing@gmail.com" and password "SEe2fRsr"
```
Comment (amstone326) [inline features/install_from_list.feature]: This is a user I created to be used only for this test; it doesn't have any privileges whatsoever and only belongs to these 2 domains, but I'm still not sure if it's ok to have the pw here like this. I wasn't sure how else to do it though so wanted to get other opinions.

File: features/install_from_list.feature [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	
+	# Verify that we're now in web user auth mode
+	Then I do not see view with id "edit_domain"
+	Then I see view with id "edit_email"
+
+	# Switch back to mobile auth view
+	Then I press view with class "android.widget.Switch" 
+
+	# Test getting app list for a mobile user
+	Then I auth for install from list with username "test" and domain "commcare-tests" and password "123"
```
Comment (ctsims) [inline features/install_from_list.feature]: Hey, not super comfortable with this being in code, we need to change this password and find a way to externalize

**Comment by wpride** [comment]
Doesn't look like calabash offers any built-in local properties feature so we might have to build this out ourselves. @ctsims would making this repository private be sufficient?

**Comment by ctsims** [comment]
The custom steps thing uses Ruby, right? Ruby should be able to read environmental variables, which we should be able to use, I would have thought? That's a common way to encode this stuff

**Comment by ctsims** [comment]
@wpride above

**Comment by wpride** [comment]
@amstone326 I added support for local.properties in I think the most Ruby-idiomatic way. You'll need to run the steps added to the README and want to update the passwords since the current ones are now public. We'll also need to add this file on Jenkins once merged.

**Comment by amstone326** [comment]
@wpride This is super cool, thank you!!

**Comment by amstone326** [comment]
@wpride Ok, the properties file is added to the ui-tests-pipeline workspace. But anyone who wants to runs this locally also needs to have a version of the properties file with the exact credentials for the commcare.mobile.testing@gmail.com web user (as opposed to for any other web user of their choosing, because the behavior I test for is specific to that user). So I guess I'll just give the password to you and Clayton offline

**Comment by amstone326** [comment]
@ctsims This is ready again now, if you want to take a look at what Will did to resolve the password issue.

**Comment by wpride** [comment]
@amstone326 Did you want this to run on AWS? If so should be added to the Jenkinsfile tag list

**Comment by amstone326** [comment]
@wpride Thanks, updated

**Pull request merged**